### PR TITLE
Add test for `umfPoolMallocUsableSize` to improve coverage of `base_alloc_global.c`

### PR DESCRIPTION
**Pull request by staniewzki**
### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Comment by ldorau** [comment]
Rebase please

File: test/poolFixtures.hpp [hunk 1]
```diff
@@ -456,4 +456,30 @@ TEST_P(umfPoolTest, allocMaxSize) {
...
 }
 
+#if defined(__has_feature)
+#if __has_feature(address_sanitizer)
+#define __SANITIZE_ADDRESS__
+#endif
+#endif
+
+#ifndef __SANITIZE_ADDRESS__
+// Sanitizer replaces malloc_usable_size implementation with it's own,
```
Comment (lukaszstolarczuk) [inline test/poolFixtures.hpp:466]: `it's` -> `its`

File: test/poolFixtures.hpp [hunk 2]
```diff
@@ -456,4 +456,30 @@ TEST_P(umfPoolTest, allocMaxSize) {
...
+// what results in a failure.
+
+TEST_P(umfPoolTest, mallocUsableSize) {
+    for (size_t allocSize : {32, 48, 1024, 8192}) {
+        char *ptr = static_cast<char *>(umfPoolMalloc(pool.get(), allocSize));
```
Comment (lukaszstolarczuk) [inline test/poolFixtures.hpp:471]: why cast to `char *`?

File: test/poolFixtures.hpp [hunk 3]
```diff
@@ -456,4 +456,30 @@ TEST_P(umfPoolTest, allocMaxSize) {
...
+        size_t result = umfPoolMallocUsableSize(pool.get(), ptr);
+        EXPECT_TRUE(result == 0 or result >= allocSize);
```
Comment (lukaszstolarczuk) [inline test/poolFixtures.hpp:473]: `or` -> `||`

**Review by lukaszstolarczuk — COMMENTED** [review]
is this supposed to be still a draft?

File: test/poolFixtures.hpp (hunk 2 above)
Comment (staniewzki) [reply test/poolFixtures.hpp:471]: I just wanted to have a byte array there. Would you like `uint8_t` better, or maybe some wider type?

File: test/poolFixtures.hpp [hunk 4]
```diff
@@ -451,4 +451,18 @@ TEST_P(umfPoolTest, realloc_compliance) {
...
 
+TEST_P(umfPoolTest, mallocUsableSize) {
+    for (size_t allocSize : {32, 48, 1024, 8192}) {
+        char *ptr = static_cast<char *>(umfPoolMalloc(pool.get(), allocSize));
+        size_t result = umfPoolMallocUsableSize(pool.get(), ptr);
+        EXPECT_TRUE(result == 0 or result >= allocSize);
+
+        for (size_t i = 0; i < result; i++) {
+            ptr[i] = 123;
+        }
```
[the remark this answers is not in the document]
Comment (staniewzki) [reply test/poolFixtures.hpp:474]: As discussed: I think that introducing this pattern in a single place in a codebase makes code overcomplicated for little gain (as it should be clear that this test is failing either way if it segfaults)

File: test/poolFixtures.hpp (hunk 2 above)
Comment (lukaszstolarczuk) [reply test/poolFixtures.hpp:467]: no casting? just `void *`?

**Review by lukaszstolarczuk — APPROVED** [review]

File: test/poolFixtures.hpp [hunk 5]
```diff
@@ -456,4 +457,24 @@ TEST_P(umfPoolTest, allocMaxSize) {
     ASSERT_EQ(ptr, nullptr);
 }
 
+TEST_P(umfPoolTest, mallocUsableSize) {
+#ifdef __SANITIZE_ADDRESS__
+    GTEST_SKIP()
+        << "This test is invalid with AddressSanitizer instrumentation";
```
Comment (PatKamin) [inline test/poolFixtures.hpp:463]: Why is this test invalid with ASAN?

Comment (lukaszstolarczuk) [reply test/poolFixtures.hpp:463]: och, right, @staniewzki, pls add back the comment that was here before

File: test/poolFixtures.hpp (hunk 2 above)
Comment (staniewzki) [reply test/poolFixtures.hpp:467]: But how to write then?

File: test/poolFixtures.hpp [hunk 6]
```diff
@@ -456,4 +457,24 @@ TEST_P(umfPoolTest, allocMaxSize) {
...
+#endif
+
+    for (size_t allocSize : {32, 48, 1024, 8192}) {
+        char *ptr = static_cast<char *>(umfPoolMalloc(pool.get(), allocSize));
+        size_t result = umfPoolMallocUsableSize(pool.get(), ptr);
+        ASSERT_TRUE(result == 0 || result >= allocSize);
```
Comment (PatKamin) [inline test/poolFixtures.hpp:469]: Why do you accept `result == 0`? This seems like an unwanted value?

**Review by PatKamin — APPROVED** [review]

**Pull request merged**